### PR TITLE
Edit Introduction

### PR DESCRIPTION
**Pull request by smart-alek**
- add/remove commas
- add semicolons
- add dashes to "trade off"
- restyle a few spots

### CODE REVIEW

File: book/index.md [hunk 1]
```diff
@@ -3,23 +3,23 @@ title: Practical Algorithms and Data Structures
 layout: chapter.html
 ---
 
-This book is a practical—and we hope, entertaining—introduction to some
-of the most important data structures and algorithms in computer
+This book is a practical—and, we hope, entertaining—introduction to some
+of the most important algorithms and data structures in computer
```
Comment (smart-alek) [inline book/index.md:7]: Keep the order of these guys consistent across the book's title, chapter and chapter's content.

File: book/index.md [hunk 2]
```diff
@@ -28,17 +28,16 @@ Those who understand trees in depth are more likely to identify tree
...
-One trade off that *we* have made is to sacrifice breadth for the sake
-of depth. We cover fewer data structures and algorithms than a typical
-introductory text may cover, in order to focus more on what we consider
-to be important. For instance we do not cover sorting algorithms at all—
-a topic that rarely arises in practice— and instead spend more time
+One trade-off that *we* have made is to sacrifice breadth for the sake
+of depth; we cover fewer data structures and algorithms than a typical
+introductory text may cover in order to focus more on what we consider
+to be important. For instance, we do not cover sorting algorithms,
+a topic that rarely arises in practice, at all; instead, we spend more time
```
Comment (smart-alek) [inline book/index.md:39]: This could comfortably be a colon as well.

File: book/index.md [hunk 3]
```diff
@@ -28,17 +28,16 @@ Those who understand trees in depth are more likely to identify tree
 shapes in their own data, unlocking superior solutions to their
 problems. Those who count Big-O analysis as a tool in their toolkits not
 only have a better vocabulary for comparing alternative strategies, they
-are also more cognizant overall of the trade offs we implicitly make
-every day. Those who see recursion everywhere see recursion everywhere
+are also more generally cognizant of the trade-offs we implicitly make
+every day. Those who see recursion everywhere see recursion everywhere...
 :)
```
Comment (smart-alek) [inline book/index.md:33]: Now that I see this again, I think you should probably toss the smiley; the ellipsis is appropriately creepy, I think.

Comment (ozan) [inline book/index.md:33]: I agree... :)

**Comment by ozan** [comment]
@aleksharma12 this is all awesome, thanks!

**Pull request merged**